tests/Schema/ComparatorTest.php

```diff
@@ -1286,4 +1286,34 @@ public function testForeignKeyRemovalWithRenamedLocalColumn(): void
         self::assertCount(1, $actual->changedTables['table2']->addedForeignKeys, 'FK to table3 should be added.');
         self::assertEquals('table3', $actual->changedTables['table2']->addedForeignKeys[0]->getForeignTableName());
     }
+
+    public function testWillNotProduceSchemaDiffOnTableWithAddedCustomSchemaDefinition(): void
+    {
+        $fromSchema = new Schema(
+            [
+                new Table(
+                    'a_table',
+                    [
+                        new Column('is_default', Type::getType('string')),
+                    ]
+                ),
+            ]
+        );
+        $toSchema   = new Schema(
+            [
+                new Table(
+                    'a_table',
+                    [
+                        new Column('is_default', Type::getType('string'), ['columnDefinition' => 'ENUM(\'default\')']),
```

Comment:
The root cause for `columnDefinition` being incorrectly picked up by the schema diffing is that:

 * `Comparator#diffColumn()` does compute the difference between `Column`s, but skips `columnDefinition`
 * `Comparator#columnsEqual()` is consistent with `Comparator#diffColumn()` when **no `$platform` is set**
 * `Comparator#columnsEqual()` reports a diff when `$platform` is set, but `diffColumn()` does not tell us what changed

This is really hard to solve, because:

 * `Platform#columnsEqual()` would need to understand `columnDefinition` to operate (we always historically bailed out there, on purpose)
 * `Comparator#diffColumn()` should probably not report a diff for `columnDefinition` anyway, as we don't know how to deal with it